src/subprocess-posix.cc

```diff
@@ -29,6 +29,18 @@ extern char** environ;
 
 #include "util.h"
 
+#define HANDLE_EAGAIN(x) ({ \
```

Comment:
This explicitly uses a non-standard GCC language extension, which is not available on all compilers that might be used for subprocess-posix.cc, and circumvents the compiler's decision making for inlining the function call (or not) as it thinks is best.

Why not make this a function, instead?

Or, since you're only using this macro in exactly one place, just put the code there directly.

Reply:
As an alternative thought on the matter, this is basically what C++11 lambda's are for (immediately invoked function expression). I would be more enthusiastic about this change if it used a C++11 lambda instead of a non-standard GCC extension.

However, you're going to need to convince the project maintainers to use C++11 or newer if you want to go that route.

Reply:
I'll push it into a separate function.